src/core/drive/page_renderer.ts

```diff
@@ -50,8 +51,8 @@ export class PageRenderer extends Renderer<HTMLBodyElement, PageSnapshot> {
     return this.newSnapshot.element
   }
 
-  mergeHead() {
-    this.copyNewHeadStylesheetElements()
+  async mergeHead() {
+    await this.copyNewHeadStylesheetElements()
     this.copyNewHeadScriptElements()
     this.removeCurrentHeadProvisionalElements()
     this.copyNewHeadProvisionalElements()
```

Comment:
This will wait for all stylesheets to be fetched before starting to fetch new scripts or fetch `<link rel=preload>`. This is not how browsers behave so I would suggest awaiting later, maybe this way:
```suggestion
    const newStylesheetElements = this.copyNewHeadStylesheetElements()
    this.copyNewHeadScriptElements()
    this.removeCurrentHeadProvisionalElements()
    this.copyNewHeadProvisionalElements()
    await newStylesheetElements
```